client/src/annoMatrix/loader.ts

```diff
@@ -281,18 +281,62 @@ export default class AnnoMatrixLoader extends AnnoMatrix {
       result.colIndex.labels()
     );
 
-    if (field === "obs") {
-      /* cough, cough - see comment on the function called */
-      _normalizeCategoricalSchema(
-        this.schema.annotations.obsByName[query],
-        result.col(query)
-      );
-    }
+    result = _responseTypeNormalization(field, query, this.schema, result);
 
     return [whereCacheUpdate, result];
   }
 }
 
+// @ts-expect-error ts-migrate(7006)
+function _responseTypeNormalization(field, query, schema, response) {
+  /*
+  Schema-driven type normalization - there are a number of assumptions the front-end
+  makes about data typing, eg, that the schema will contain all categories in a categorical
+  column, that booleans are a JS array of true/false, etc.
+
+  The OTA format does not follow precisely the same conventions. This routine implements
+  the front-end conventions given the schema and an OTA data column.
+  */
+  if (field === "obs" || field === "var") {
+    response = _booleanCast(field, query, schema, response);
+  }
+  if (field === "obs") {
+    /* 
+    Cough, cough - see comment on the function called.
```

Comment:
I dont think I understand this comment at all

Reply:
lolz, yes.  Old comment from when the code was first created in 2020 (perhaps expressing some discomfort with the approach).

It is a pointer to the comment at the head of the _normalizeCategoricalSchema() function, which explains what it does.

Here is the comment (from schema.js):
```
/*
  Ensure all enum schema types have a categories array, that
  the categories array contains all unique values in the data
  array, AND that the array is sorted.
  Note that the back-end will not always set this hint, so we
  must assume it may be incorrect and/or missing.
  */
```

Reply:
I cleaned up the comment.